src/components/HomeSearchBox/index.tsx

```diff
@@ -1,42 +1,110 @@
-import React, { useState } from 'react'
+import React, { useState, useEffect } from 'react'
 import { useRouter } from 'next/router'
 import useTranslation from 'next-translate/useTranslation'
 
 const HomeSearchBox = () => {
+  const [searchValue, setSearchValue] = useState<string>('')
+  const [active, setActive] = useState(0)
+  const [filteredSuggestion, setFilteredSuggestion] = useState<string[]>([])
+  const [isSuggestionShown, setSuggestionShown] = useState(false)
   const { t } = useTranslation('home')
   const router = useRouter()
-  const [searchValue, setSearchValue] = useState<string>('')
+
+  useEffect(() => {
+    const handleMouseClick = (event: MouseEvent) => {
+      setSuggestionShown(false)
+    }
+
+    document.addEventListener('mousedown', handleMouseClick)
+
+    return () => {
+      document.removeEventListener('mousedown', handleMouseClick)
+    }
+  }, [])
 
   const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setSearchValue(event.target.value)
+    //TODO: This below data should be obtained from database
+    const suggestions: string[] = ['Alabama', 'Alaska', 'American Samoa', 'Bubble Holmes', 'Contra Hits', 'Amazon']
+
+    const {
+      currentTarget: { value }
+    } = event
+    const newFilteredSuggestionSuggestions = suggestions.filter(
```

Comment:
Please rename this function
```suggestion
    const filteredSuggestions = suggestions.filter(
```